pulsar-io/core/src/main/java/org/apache/pulsar/io/core/utils/Backoff.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.io.core.utils;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.time.Clock;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import lombok.Data;
+
+
+/**
+ * All variables are in TimeUnit millis by default.
+  */
+@Data
+public class Backoff {
```

Comment:
this class is becoming part of the Public API.
please annotate it with Stability annotations.

Can't you move this class to the Kinesis connector module ?

I had a similar discussion with @vroyer that wanted to add a SslConfig class in the public Pulsar IO package.

those additions should be done with care because once we add something to the public API we have to maintain it in the future

Reply:
Yes, I think you are right, it seems that not all connectors need this class. I move it to the project Kinesis connector.